### PR TITLE
Bump Knative to v0.23.0

### PR DESCRIPTION
**Pull request by boramalper**
Signed-off-by: Bora M. Alper <bora@boramalper.org>

### CODE REVIEW

**Comment by boramalper** [comment]
**Does NOT Work Yet!**

Using [ease-lab/serving@v0.23_vhive](https://github.com/ease-lab/serving/tree/v0.23_vhive), I have tried to update Knative for vHive to v0.23.0 ([the instructions](https://github.com/ease-lab/vhive/wiki/TMP:-How-to-Update-Knative-for-vHive)).

When I set up a vHive single-node stock-only Knative node (using the new YAML files that I have generated, see [boramalper/vhive@v0.23_vhive](https://github.com/boramalper/vhive/tree/v0.23_vhive) and [`generate-yamls.sh` logs](https://gist.github.com/boramalper/660b005842e00a7f4d96421dc13f52bc)), I still get Knative v0.21.0 installed:

```bash
$ kubectl get namespace knative-serving -o 'go-template={{index .metadata.labels "serving.knative.dev/release"}}'
v0.21.0
$ kubectl get namespace keventing -o 'go-template={{index .metadata.labels "eventing.knative.dev/release"}}'}}'
v0.21.0
$ kn version
Version:      v20210601-local-3cc608f
Build Date:   2021-06-01 13:32:05
Git Revision: 3cc608f
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.23.0)
* Eventing
  - sources.knative.dev/v1 (knative-eventing v0.23.0)
  - eventing.knative.dev/v1 (knative-eventing v0.23.0)
```

**Comment by boramalper** [comment]
I have found that there is a mismatch between the tags of images published and the configuration files generated by `generate-yamls.sh`:

**`./configs/knative_yamls/serving-hpa.yaml`**:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: autoscaler-hpa
  namespace: knative-serving
  labels:
    serving.knative.dev/release: devel
    autoscaling.knative.dev/autoscaler-provider: hpa
spec:
  # [...]
  template:
    spec:
      # [...]
      containers:
        - name: autoscaler-hpa
          # This is the Go import path for the binary that is containerized
          # and substituted here.
          image: docker.io/vhiveease/autoscaler-hpa-85c0b68178743d74ff7f663a72802ceb:ad1052f88caa83f05335d2d7cd56ca3598991387986351727281715bf785f4c6
```

whereas the published image has been tagged `latest`:

image

Investigating why tagging goes wrong, and how to fix it or do it correctly in the first place.

**Comment by shyamjesal** [comment]
`latest` is a pseudo tag. The uploaded images do have their own tag apart from latest. Try finding that and then compare.

**Comment by boramalper** [comment]
I tried twice (and yet another time with @MBaczun) today and all worked fine---Michal experienced some bugs with their Knative setup that I could not replicate and believe are transient.

I believe that the issues I experienced yesterday was due to using the wrong repo & branch...

Would any of you (@shyamjesal or @ustiugov) be able to try setting it up yourselves and confirm that it works? I am confident that it does, but would appreciate a third confirmation if possible.

## Testing Stock Knative Images
```bash
git clone --branch=v0.23_vhive https://github.com/boramalper/vhive
cd vhive
./scripts/cloudlab/setup_node.sh stock-only
sudo containerd  # in the background
./scripts/cluster/create_one_node_cluster.sh stock-only
# wait for the containers to boot up using 
watch kubectl get pods -A
```

**[Checking the version of your Knative components](https://knative.dev/docs/install/check-install-version/):**
```bash
kubectl get namespace knative-serving -o 'go-template={{index .metadata.labels "serving.knative.dev/release"}}'
kubectl get namespace knative-eventing -o 'go-template={{index .metadata.labels "eventing.knative.dev/release"}}'
```
Both shall return `v0.23.0`.

**Comment by ustiugov** [comment]
thanks! we have the CI for that :) 

Can you please push your branch to the upstream repo and run the "Nightly integration tests" on that branch. if tests fail, I'll take a closer look

File: configs/knative_yamls/serving-default-domain.yaml [hunk 1]
```diff
@@ -33,8 +33,8 @@ spec:
         - name: default-domain
           # This is the Go import path for the binary that is containerized
           # and substituted here.
-          image: docker.io/vhiveease/default-domain-dd6f55160290a90018ba318b624be12e:998c8975f56b3b45059fd709d3776062b51cb1aa84921d4d7a85c0315d9cd0f0
-          args: ["-magic-dns=nip.io"]
+          image: docker.io/vhiveease/default-domain-dd6f55160290a90018ba318b624be12e@sha256:b9afac8d6bba3a4d055b7b636bcc6d6e61feb1e3936728cea09dabaea50648ca
+          args: ["-magic-dns=sslip.io"]
```
Comment (ustiugov) [inline configs/knative_yamls/serving-default-domain.yaml:37]: need to replace `xip.io` in the CRI tests (grep in the `cri/` folder) with sslip.io

**Review by ustiugov — CHANGES_REQUESTED** [review]
Minor things (and hope the CRI tests would pass)

- [ ] need a simple fix in the CRI tests in the `cri` folder
- [ ] add a line to the Changelog, we gonna do 1.3 vHive release next week

**Review by ustiugov — DISMISSED** [review]

**Review by ustiugov — DISMISSED** [review]

**Comment by ustiugov** [comment]
@shyamjesal can you please take a look at the artifact? Bora can troubleshoot with some of your guidance
https://github.com/ease-lab/vhive/actions/runs/902388125

**Comment by boramalper** [comment]
Based on https://github.com/ease-lab/serving/pull/5.

vHive Nightly Integration Tests results: https://github.com/ease-lab/vhive/actions/runs/914303428

Only [CRI tests (nightly) (-dbg -snapshots -upf)](https://github.com/ease-lab/vhive/runs/2762712157?check_suite_focus=true) has failed.

**Review by ustiugov — COMMENTED** [review]
lgtm, let's wait for the manual check

**Comment by boramalper** [comment]
Manual check is fine too. See https://asciinema.org/a/Y6I1HWdDLeJlvf0T191wOWxxl

**Review by ustiugov — APPROVED** [review]

**Pull request merged**